### PR TITLE
Enhance grpc-test-utils and grpc-reverse-proxy to use Resources

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
This is the second of four PRs in which https://github.com/digital-asset/daml/commit/6ea70c4b45d55cbf62caa3cc03dd70310ca79b87
has been broken up to facilitate review.

The endgame is to have the non-repudiation prototype merged. The
grpc-test-utils and grpc-server-proxy libraries have been
enhanced to use the ResouceOwner/Resource abstraction to handle
the lifecycle of components, making resource management easier
when testing.

Part of #8598

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: libs-scala/grpc-test-utils/src/main/scala/com/daml/grpc/test/GrpcServer.scala [hunk 1]
```diff
@@ -29,34 +98,20 @@ trait GrpcServer { this: AsyncFlatSpec =>
...
-
-}
-
-object GrpcServer {
+    val channelBuilder = InProcessChannelBuilder.forName(serverName)
+    val testResult =
+      for {
+        _ <- Resources.forServer(serverBuilder, 5.seconds).acquire()
+        channel <- Resources.forChannel(channelBuilder, 5.seconds).acquire()
+        result <- Resources.forFuture(() => test(channel)).acquire()
```
Comment (cocreature) [inline libs-scala/grpc-test-utils/src/main/scala/com/daml/grpc/test/GrpcServer.scala:111]: Maybe `use` would fit here better?

**Review by cocreature — APPROVED** [review]
Nice, thanks!

File: libs-scala/grpc-test-utils/src/main/scala/com/daml/grpc/test/GrpcServer.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply libs-scala/grpc-test-utils/src/main/scala/com/daml/grpc/test/GrpcServer.scala:111]: Well-spotted, thanks!

Comment (stefanobaghino-da) [reply libs-scala/grpc-test-utils/src/main/scala/com/daml/grpc/test/GrpcServer.scala]: Addressed by https://github.com/digital-asset/daml/pull/8606/commits/e29ecb54d06666097e5b15e1c1df9fa1db65d0ba

**Pull request merged**